azavea/raster-foundry
# New histogram with clipping

## Overview
Update histogram component with clipping and add to color correct

Remove old clipping slider

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
![image](https://cloud.githubusercontent.com/assets/4392704/26217187/8b561d9e-3bd4-11e7-8e6d-55d93bc7a9b9.png)
![image](https://cloud.githubusercontent.com/assets/4392704/26217167/7c543c5e-3bd4-11e7-88f9-a04ea5bf6e73.png)
![image](https://cloud.githubusercontent.com/assets/4392704/26217174/80a19ebe-3bd4-11e7-9010-c0ba4a37e79b.png)
![image](https://cloud.githubusercontent.com/assets/4392704/26217181/8619e3d8-3bd4-11e7-8668-541212ac4d8d.png)

### Notes

* The eyedropper buttons don't do anything right now. 
* The histogram reflects clipping changes. In the future we may want to change this.

## Testing Instructions

* Go to the advanced color correction view
* Click the histogram and confirm that it moves the closest clipping to that value
* Verify that you can set each channel independently
* Verify that values are conservatively set (e.g. if red min/max are the same as the overall min/max, set them to null) in the request and do not not show on the plot unless they are overriding.
* Verify that changing the overall clipping resets all channel clipping to the same values
* Verify that values persist across reloads etc
* Verify that the histogram updates when color correction changes are made
* Verify that changes using the textboxes are debounced by 500ms (so holding down the down arrow doesn't send a new request for every value)
* Verify that the histogram mouseover text and (i) icon tooltip makes sense

Closes #1645
